plugins/utils/mini.nix

```diff
@@ -1,27 +1,18 @@
 {
   lib,
-  helpers,
-  config,
-  pkgs,
   ...
 }:
-with lib;
-let
-  cfg = config.plugins.mini;
-in
-{
-  options.plugins.mini = {
-    enable = mkEnableOption "mini.nvim";
+lib.nixvim.neovim-plugin.mkNeovimPlugin {
+  name = "mini";
+  originalName = "mini.nvim";
 
-    package = lib.mkPackageOption pkgs "mini.nvim" {
-      default = [
-        "vimPlugins"
-        "mini-nvim"
-      ];
-    };
+  maintainers = [ lib.maintainers.khaneliman ];
 
-    modules = mkOption {
-      type = with types; attrsOf attrs;
+  package = "mini-nvim";
+
+  extraOptions = {
+    modules = lib.mkOption {
+      type = with lib.types; attrsOf (attrsOf anything);
       default = { };
       description = ''
```

Comment:
The description and example here are probably due some love.

In particular, we could link to https://github.com/echasnovski/mini.nvim?tab=readme-ov-file#modules and maybe expand the example to be more comprehensive?

Reply:
Ah, yeah. I'll update.